### ISSUE TITLE
Cannot pass arguments beginning with a hyphen in --args

### ISSUE THREAD

**Issue by Pascal-Delange** [google-github-actions/deploy-cloudrun#444]
### TL;DR

I cannot pass arguments starting with a hyphen (say, -run_server) in the --args. Indeed, it translates it to the gcloud command `--args -run_server` which then complains about no value found in --args. The correct syntax here would be to translate it to `--args=run_server` which works.

### Expected behavior

`flags: --args=-server`  gets translated to the gcloud argument `--args -server`, which fails because of the hyphen

### Observed behavior

The "=" between --args and the actual args should remain there

### Action YAML

```yaml
- name: Deploy server
     uses: "google-github-actions/deploy-cloudrun@v1"
     with:
       service: service-name
       image: my-image
       region: my-region
       flags: --args="-server"
```

### Additional information

In the end I just fell back to standard gcloud commands, since I also need to deploy a Cloud Run job anyway.

**Comment by sethvargo** [comment]
Hi @Pascal-Delange - can you please try using the latest version of deploy-cloudrun (v2 as of now) and supply the GitHub Actions debug logs?

**Comment by sethvargo** [comment]
Closing due to lack of response. Please comment or open a new issue if this is still happening.

**Comment by Pascal-Delange** [comment]
Hi @sethvargo ,

I tried to run it with v2, to the same effect. Below are the logs: 
```
2024-04-23T09:42:17.3999781Z ##[group]Run google-github-actions/deploy-cloudrun@v2
2024-04-23T09:42:17.4000508Z with:
2024-04-23T09:42:17.4000919Z   service: marble-backend
2024-04-23T09:42:17.4001675Z   image: europe-west1-docker.pkg.dev/marble-infra/marble/marble-backend:latest
2024-04-23T09:42:17.4002388Z   region: europe-west1
2024-04-23T09:42:17.4002921Z   flags: --args="-server"
2024-04-23T09:42:17.4003395Z   skip_default_labels: false
2024-04-23T09:42:17.4003946Z   no_traffic: false
2024-04-23T09:42:17.4004382Z env:
2024-04-23T09:42:17.4005063Z   IMAGE: europe-west1-docker.pkg.dev/marble-infra/marble/marble-backend:latest
2024-04-23T09:42:17.4005881Z   CLOUDSDK_METRICS_ENVIRONMENT: github-actions-setup-gcloud
2024-04-23T09:42:17.4006587Z   CLOUDSDK_METRICS_ENVIRONMENT_VERSION: 1.1.1
2024-04-23T09:42:17.4007529Z   CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE: /home/runner/work/marble-backend/marble-backend/gha-creds-484988367281c453.json
2024-04-23T09:42:17.4008891Z   GOOGLE_APPLICATION_CREDENTIALS: /home/runner/work/marble-backend/marble-backend/gha-creds-484988367281c453.json
2024-04-23T09:42:17.4010017Z   GOOGLE_GHA_CREDS_PATH: /home/runner/work/marble-backend/marble-backend/gha-creds-484988367281c453.json
2024-04-23T09:42:17.4010941Z   CLOUDSDK_CORE_PROJECT: tokyo-country-381508
2024-04-23T09:42:17.4011541Z   CLOUDSDK_PROJECT: tokyo-country-381508
2024-04-23T09:42:17.4012170Z   GCLOUD_PROJECT: tokyo-country-381508
2024-04-23T09:42:17.4012708Z   GCP_PROJECT: tokyo-country-381508
2024-04-23T09:42:17.4013578Z   GOOGLE_CLOUD_PROJECT: tokyo-country-381508
2024-04-23T09:42:17.4014188Z ##[endgroup]
2024-04-23T09:42:18.3449601Z Successfully authenticated
2024-04-23T09:42:18.3453038Z Running: gcloud run deploy marble-backend --quiet --image europe-west1-docker.pkg.dev/marble-infra/marble/marble-backend:latest --update-labels ^2e^managed-by=github-actions2ecommit-sha=0eddb09a9ae08cf22ff4dfd51dbcb9a98207b768 --format json --region europe-west1 --args "-server"
2024-04-23T09:42:25.8566974Z ##[error]google-github-actions/deploy-cloudrun failed with: failed to execute gcloud command `gcloud run deploy marble-backend --quiet --image europe-west1-docker.pkg.dev/marble-infra/marble/marble-backend:latest --update-labels ^2e^managed-by=github-actions2ecommit-sha=0eddb09a9ae08cf22ff4dfd51dbcb9a98207b768 --format json --region europe-west1 --args "-server"`: Deploying container to Cloud Run service [marble-backend] in project [tokyo-country-381508] region [europe-west1]
Deploying...
Creating Revision..........................................failed
Deployment failed
ERROR: (gcloud.run.deploy) Revision 'marble-backend-00410-x27' is not ready and cannot serve traffic. The user-provided container failed to start and listen on the port defined provided by the PORT=80 environment variable. Logs for this revision might contain more information.
```
In particular, it's still translating to `gcloud ... --args "-server"` instead of `gcloud ... --args="-server"`
For what it's worth, it's no longer really blocking for me (when I first had the issue I switched to raw `gcloud` commands, and now those flags are defined in a terraform file for us. 
I just thought it may still be useful feedback.
